envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt

```diff
@@ -3,30 +3,39 @@ package pl.allegro.tech.servicemesh.envoycontrol.services
 typealias ServiceName = String
 
 data class ServicesState(
-    val serviceNameToInstances: Map<ServiceName, ServiceInstances> = emptyMap()
+    val serviceNameToInstances: MutableMap<ServiceName, ServiceInstances> = mutableMapOf()
 ) {
     operator fun get(serviceName: ServiceName): ServiceInstances? = serviceNameToInstances[serviceName]
 
     fun hasService(serviceName: String): Boolean = serviceNameToInstances.containsKey(serviceName)
     fun serviceNames(): Set<ServiceName> = serviceNameToInstances.keys
     fun allInstances(): Collection<ServiceInstances> = serviceNameToInstances.values
 
-    fun remove(serviceName: ServiceName): ServicesState {
-        // TODO: https://github.com/allegro/envoy-control/issues/11
-        return change(ServiceInstances(serviceName, instances = emptySet()))
+
+    fun remove(serviceName: ServiceName): Boolean {
+        return if (serviceName in serviceNameToInstances) {
```

Comment:
Method remove returns previous value or null. You dont need to use if.
Please consider:
`serviceNameToInstances.remove(serviceName) != null`

Reply:
thanks for pointing it out. Fixed